src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy

```diff
@@ -12,21 +15,28 @@ import javax.ws.rs.core.Response
  */
 @Path('/')
 class InfoResource extends Resource {
+    Logger logger = LoggerFactory.getLogger(InfoResource.class)
+
     private Info info = new Info()
 
     public InfoResource() {
-        def stream = this.class.getResourceAsStream('/build.properties')
-        if (stream == null) {
-            throw new Exception("couldn't load build.properties")
-        }
+        try {
+            def stream = this.class.getResourceAsStream('/build.properties')
+            if (stream == null) {
+                throw new Exception("couldn't load build.properties")
+            }
 
-        def properties = new Properties()
-        properties.load(stream)
+            def properties = new Properties()
+            properties.load(stream)
+
+            info.name = properties.get('name')
+            info.time = Long.parseLong(properties.getProperty('time'))
+            info.commit = properties.get('commit')
+            info.documentation = properties.get('documentation')
+        } catch (Exception e) {
+            logger.error("Resource initialization exception when loading InfoResource build.properties", e)
```

Comment:
> Hmm, I just realized that this is not the resource method.

No, but we could hold onto the error and return it when somebody does try to GET the info resource. I like that idea. It's better than returning a zeroed Info object.

Although tbh i still think we should just crash here. There's really no good reason why build.properties should be missing.